lib/chai/core/assertions.js

```diff
@@ -992,23 +992,38 @@ module.exports = function (chai, _) {
    *     expect({ foo: 1, bar: 2 }).to.have.any.keys('foo');
    *     expect({ foo: 1, bar: 2 }).to.contain.any.keys('bar', 'baz');
    *     expect({ foo: 1, bar: 2 }).to.contain.any.keys(['foo']);
+   *     expect({ foo: 1, bar: 2 }).to.contain.any.keys({'foo': 6});
    *     expect({ foo: 1, bar: 2 }).to.have.all.keys(['bar', 'foo']);
+   *     expect({ foo: 1, bar: 2 }).to.have.all.keys({'bar': 6, 'foo', 7});
    *     expect({ foo: 1, bar: 2, baz: 3 }).to.contain.all.keys(['bar', 'foo']);
+   *     expect({ foo: 1, bar: 2, baz: 3 }).to.contain.all.keys([{'bar': 6}}]);
+   *
    *
    * @name keys
    * @alias key
-   * @param {String...|Array} keys
+   * @param {String...|Array|Object} keys
    * @api public
    */
 
   function assertKeys (keys) {
     var obj = flag(this, 'object')
       , str
-      , ok = true;
+      , ok = true
+      , mixedArgsMsg = "keys: if first argument is instanceof Array|Object, must be only argument."
+
+    if (keys instanceof Array || keys instanceof Object) {
+      if (arguments.length > 1) {
+        throw (new Error(mixedArgsMsg));
```

Comment:
The 'throw' was mentioned in the bug, so I added it.  I would think it would be very strange to do `keys(Array, string...)` as dev, but it could happen by accident.  Low-cost to check :)